src/engine/Util/Detector.ts

```diff
@@ -5,9 +5,117 @@ module ex {
     * Excalibur internal feature detection helper class
     */
    export class Detector {
-      
+
+      private _features: Object = null;
+
       public failedTests: string[] = [];
+
+      public constructor() {
+         this._features = this._loadBrowserFeatures();
+      }
+
+      /**
+       * Returns a map of currently supported browser features. This method
+       * treats the features as a singleton and will only calculate feature
+       * support if it has not previously been done.
+       * 
+       * @return {Object} the list of supported features
```

Comment:
I know this automatically generated by VSCode, but you can technically leave off the return type info--TypeDoc (our doc generator) can infer it via TypeScript compiler.